chutney/server/src/main/java/fr/enedis/chutney/index/infra/config/OnDiskIndexConfig.java

```diff
@@ -7,38 +7,74 @@
 
 package fr.enedis.chutney.index.infra.config;
 
+import static fr.enedis.chutney.tools.file.FileUtils.cleanFolder;
 import static fr.enedis.chutney.tools.file.FileUtils.initFolder;
 
 import fr.enedis.chutney.index.infra.CustemChutneyAnalyzer;
 import java.io.IOException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.Arrays;
 import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.index.CorruptIndexException;
 import org.apache.lucene.index.IndexWriter;
 import org.apache.lucene.index.IndexWriterConfig;
 import org.apache.lucene.index.TieredMergePolicy;
 import org.apache.lucene.store.Directory;
 import org.apache.lucene.store.FSDirectory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class OnDiskIndexConfig implements IndexConfig {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OnDiskIndexConfig.class);
+
     private final IndexWriter indexWriter;
     private final Directory indexDirectory;
     private final Analyzer analyzer;
 
     public OnDiskIndexConfig(String indexDir, String indexName) {
+        Path path;
         try {
-            Path path = Paths.get(indexDir, indexName);
+            path = Paths.get(indexDir, indexName);
```

Comment:
extractMethods to have something like this : 

analyzer = new CustemChutneyAnalyzer();
indexDirectory = initFromPath(...)
indexWriter = readOrCleanIndex(...)